src/main/java/org/testng/internal/MethodHelper.java

```diff
@@ -335,6 +339,64 @@ private static String calculateMethodCanonicalName(Method m) {
     return result;
   }
 
+  public static void fixMethodsWithClass(ITestNGMethod[] methods,

```

Comment:
Cannot do this, because this method is mostly being invoked from values extracted from `org.testng.ITestClass` wherein the methods are mostly being returned as an array.